diff --git a/src/vitepress/assets/shopware-docs.png b/src/vitepress/assets/shopware-docs.png
new file mode 100644
index 0000000..ebb3c1f
Binary files /dev/null and b/src/vitepress/assets/shopware-docs.png differ
diff --git a/src/vitepress/assets/shopware-docs.svg b/src/vitepress/assets/shopware-docs.svg
new file mode 100644
index 0000000..98d5136
--- /dev/null
+++ b/src/vitepress/assets/shopware-docs.svg
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Generator: Adobe Illustrator 24.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
+<svg version="1.1" id="Ebene_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 viewBox="-2 -2 673 114" style="enable-background:new 0 0 668.3 110;" xml:space="preserve">
+<style type="text/css">
+	.st0{fill:#FFFFFF;}
+</style>
+<g id="Ebene_2_1_">
+</g>
+<g id="Ebene_1_1_">
+	<g>
+		<g id="XMLID_454_">
+			<g id="XMLID_479_">
+				<path id="XMLID_480_" class="st0" d="M150.2,55.5c-4.7-2-8.5-3.9-8.5-6.9c0-2.1,2.4-4.2,6.5-4.2c3.7,0,6.8,0.9,10.1,2.3
+					c0.9,0.4,2.3,0.5,3-0.9l1.6-3.2c0.6-1.2,0.1-2.6-1-3.5c-2.3-1.5-7.2-3.5-13.7-3.5c-11.4,0-16,6.9-16,12.8
+					c0,7.7,6,11.6,13.2,14.8c6.5,2.9,10.1,5,10.1,8.1c0,2.6-2.2,4.6-6.2,4.6c-5.6,0-8.9-1.9-11.7-3.5c-1.2-0.7-2.5-0.4-3,0.6
+					l-2.1,4.4c-0.5,1,0.1,1.9,0.8,2.4c2.6,1.8,8,4.8,15.4,4.8c10.5,0,16.8-6.4,16.8-13.7C165.3,62.3,158.8,59.2,150.2,55.5z"/>
+			</g>
+			<g id="XMLID_476_">
+				<path id="XMLID_478_" class="st0" d="M87.8,98c-4.8-3.6-11.8-6.3-19.3-9.2c-8.9-3.4-18.9-7.3-26.5-13.3
+					C33.4,68.6,29.3,60,29.3,49.1c0-9.7,4-18.1,11.7-24.1c8.6-6.8,21.6-10.3,37.7-10.3c4.4,0,8.6,0.3,12.6,0.8
+					c0.3,0,0.7-0.1,0.8-0.4c0.1-0.3,0.1-0.7-0.2-0.9C81.6,5,68.6,0,55,0C40.4,0,26.5,5.7,16.1,16.1S0,40.4,0,55
+					c0,14.7,5.7,28.6,16.1,38.9C26.5,104.3,40.4,110,55,110c11.9,0,23.1-3.7,32.8-10.8c0.2-0.1,0.3-0.4,0.3-0.6
+					C88.1,98.4,88,98.2,87.8,98z"/>
+				<path id="XMLID_477_" class="st0" d="M109.8,50c0-0.2-0.1-0.4-0.3-0.5C98,40,88.8,36.1,77.4,36.1c-6,0-10.6,1.2-13.7,3.6
+					c-2.7,2.1-4.1,4.9-4.1,8.3c0,9.4,11.5,13.7,24.9,18.7c6.9,2.6,14,5.2,20.5,8.8c0.1,0.1,0.2,0.1,0.4,0.1c0.1,0,0.2,0,0.3,0
+					c0.2-0.1,0.4-0.2,0.5-0.5c2.5-6.4,3.8-13.1,3.8-20C110.2,53.4,110,51.7,109.8,50z"/>
+			</g>
+			<path id="XMLID_473_" class="st0" d="M426.1,84.3c1.4-0.2,2.4-1.2,2.4-2.5V57.9c0-12.2-1.5-21.6-17.2-21.6c-9,0-16,2.4-16.9,2.8
+				c-1.2,0.4-1.9,1.2-1.6,2.9l1,4c0.3,1.4,1.3,2.3,2.8,2c2.5-0.7,8.1-2.3,13.7-2.3c6.1,0,8.1,3.5,7.8,11.2c0,0-4.8-1.5-9.9-1.5
+				c-12.6,0-19.6,6.8-19.6,15.8c0,10.2,6.5,13.6,15.7,13.6c7.5,0,15.3-0.1,18.5-0.3L426.1,84.3z M418,75.6
+				c-1.5,2.4-6.8,1.8-11.3,1.7c-5.1-0.1-7.6-1.9-7.6-6.2c0-5.1,4.3-8.1,10.1-8.1c4.5,0,8.3,1.3,8.9,1.7v10.9H418z"/>
+			<path id="XMLID_470_" class="st0" d="M506.3,35.7c3.1,0,5.6-2.5,5.6-5.6s-2.5-5.6-5.6-5.6s-5.6,2.5-5.6,5.6
+				S503.2,35.7,506.3,35.7z M506.3,25.3c2.6,0,4.7,2.1,4.7,4.7s-2.1,4.7-4.7,4.7c-2.6,0-4.7-2.1-4.7-4.7
+				C501.6,27.6,503.7,25.3,506.3,25.3z"/>
+			<path id="XMLID_467_" class="st0" d="M504.4,33h0.5c0.1,0,0.2-0.1,0.2-0.2v-2h1.2l1.1,2.1c0,0.1,0.1,0.1,0.2,0.1h0.6
+				c0.1,0,0.2-0.1,0.1-0.2l-1.1-2c0.7-0.3,1.2-0.9,1.2-1.6c0-1-0.8-1.7-1.8-1.7h-2.2c-0.1,0-0.2,0.1-0.2,0.2V33
+				C504.2,32.9,504.3,33,504.4,33z M505.1,28.1h1.4c0.5,0,1,0.4,1,1s-0.5,1-1,1h-1.4V28.1z"/>
+			<path id="XMLID_466_" class="st0" d="M196.6,35.7c-7.4,0-12.5,4.1-13.6,4.9v-21c0-1.1-1.1-2.2-2.2-2.2H175
+				c-1.2,0-2.2,1.1-2.2,2.2v63c0,1.2,1,2.2,2.2,2.2h5.6c1.4,0,2.3-0.9,2.3-2.2V52.2c1-2,5.1-6.8,12.3-6.8c6.1,0,9.3,4.6,9.3,12.3
+				v24.9c0,1.2,1,2.2,2.2,2.2h5.7c1.1,0,2.2-1,2.2-2.2V57.9C214.8,45.1,209.9,35.7,196.6,35.7z"/>
+			<path id="XMLID_463_" class="st0" d="M315.8,58.4c-0.6-15.9-7.2-22.6-22.6-22.6c-7.4,0-16.2,2.2-17.5,2.8
+				c-1.6,0.7-2.8,1.5-2.8,2.6v0.9v3.4v11V58v43c0,1.2,1,2.2,2.2,2.2h4.7c1.5,0,2.4-0.9,2.4-2.2V84.1c1.4,0.4,8.7,1,14.8,0.8
+				c10.9-0.3,18.2-8.2,18.8-21.5C315.9,61.4,315.8,58.4,315.8,58.4z M306,63.2c-0.2,9-2.8,13-10.4,13.4c-3.8,0.2-12.6-0.8-12.6-0.8
+				c-0.4-0.1-0.6-0.3-0.6-0.6V56.9V46.5c0-0.3,0.2-0.5,0.5-0.5c1-0.1,4.9-1.3,10.4-1.3c9.3,0,12.5,1.4,12.8,13.1
+				C306,59.3,306,61.4,306,63.2z"/>
+			<path id="XMLID_462_" class="st0" d="M383.2,36.7h-4.3c-1.2,0-2.9,0-3.6,1.4l-7.5,27h-0.2l-12.6-28.4c-0.2-0.6-0.9-1.1-2-1.1
+				h-0.7c-0.9,0-1.6,0.5-1.9,1.1l-12.7,28.4h-0.2l-7.8-26.9c-0.2-0.9-1.2-1.4-2-1.4h-6.1c-1.6,0-2.3,1.2-1.6,2.6l13.7,44.3
+				c0.2,0.7,0.9,1.2,1.9,1.2h0.6c1,0,1.6-0.5,1.8-1.1l14.4-32h0.2l14.3,32c0.3,0.6,1,1.1,2,1.1h0.4c1,0,1.9-0.5,2.1-1.2l13.4-44.3
+				C385.7,38,384.8,36.7,383.2,36.7z"/>
+			<path id="XMLID_461_" class="st0" d="M458.2,36.3c-0.2,0-0.5,0-0.7,0c-0.4,0-0.8,0-1.2,0c-2.4,0-4.8,0.1-7,0.2l0,0
+				c-4.2,0.2-6.7,0.5-8.5,0.7c0,0-1.1,0.1-2.5,0.3c-0.9,0.1-1.5,0.5-1.9,1c-0.3,0.3-0.4,0.7-0.4,1.1v0.2v0.9v3.4v9.4v29.2
+				c0,1.2,1,2.2,2.2,2.2h4.7c1.9,0,3.2-0.2,3.2-2.2V53.5l0,0v-6.7c0-0.3,0.4-0.6,0.7-0.8c1.1-0.7,3.9-1.2,9.4-1.2c0.5,0,0.9,0,1.3,0
+				c1.3,0.1,3.3,0.5,3.9,0.9c1,0.5,2.2,0.2,2.8-1.1l2.4-4C468.3,37.5,462.6,36.3,458.2,36.3z"/>
+			<path id="XMLID_458_" class="st0" d="M490.4,35.7c-14.1,0-23.4,11.3-23.4,24.5c0,13.4,9.2,24.6,23.5,24.6c7.8,0,14-2.8,17.4-5.8
+				c1.4-1.2,1-2.1,0.2-3.1l-2.7-4.3c-0.8-1.2-1.7-0.9-2.7-0.2c-2.8,2-6.1,4-11.5,4c-8.2,0-13.9-6.5-14-12.8h32.5c1,0,2-0.8,2.1-2
+				c0.1-0.8,0.2-2.4,0.2-3.2C511.8,44.8,502.6,35.7,490.4,35.7z M477.6,55c0.7-5.9,5.2-10.7,12.6-10.7c6.9,0,11.4,5,11.6,10.7H477.6
+				z"/>
+			<path id="XMLID_455_" class="st0" d="M264.6,50.2c-1.1-3-2.5-5.6-4.5-7.8c-1.9-2.2-4.2-3.8-6.8-5c-2.6-1.2-6.4-1.8-9.5-1.8
+				s-6.9,0.6-9.5,1.8c-2.6,1.2-4.9,2.8-6.8,5c-1.9,2.2-3.4,4.7-4.5,7.8c-1.1,3-1.6,6.4-1.6,10c0,3.7,0.5,7.1,1.6,10.1
+				c1.1,3,2.5,5.6,4.5,7.7c1.9,2.2,4.2,3.8,6.8,5c2.6,1.2,6.4,1.8,9.5,1.8s6.9-0.6,9.5-1.8c2.6-1.2,4.9-2.8,6.8-5s3.4-4.7,4.5-7.7
+				c1.1-3,1.6-6.3,1.6-10.1C266.2,56.6,265.7,53.3,264.6,50.2z M243.5,76.7c-8.5,0-12.7-7.2-12.7-16s4.3-16,12.7-16
+				c8.5,0,12.8,7.2,12.8,16S252.1,76.7,243.5,76.7z"/>
+		</g>
+		<g>
+			<g>
+				<path class="st0" d="M539.5,71c0-11.1,7.2-20,16.9-20c4.5,0,7.4,2.1,7.6,2.1l2.4-16.8c0.1-0.9,0.6-1.6,1.6-1.6h2.5
+					c0.8,0,1.4,0.7,1.3,1.6l-6.6,47.5c-0.1,0.9-0.8,1.2-1.5,1.2H562c-0.7,0-1-0.5-1.1-1.2l-0.3-2.1c-0.1,0-4,4.1-9.9,4.1
+					C544.2,85.7,539.5,79,539.5,71z M561,75.2l2.3-17.4c0,0-2.7-2.1-6.9-2.1c-6.2,0-11.4,6.8-11.4,14.7c0,5.3,2.4,10.5,7.2,10.5
+					S561,75.2,561,75.2z"/>
+				<path class="st0" d="M593.9,51c7,0,12.8,6.7,12.8,14.8c0,10.7-7.7,19.8-17.4,19.8c-7,0-12.8-6.8-12.8-15
+					C576.4,60,584.2,51,593.9,51z M601.6,66.2c0-5.8-3.3-10.5-8.2-10.5c-6.2,0-11.9,6.4-11.9,14.5c0,5.9,3.2,10.7,8.2,10.7
+					C595.9,80.9,601.6,74.3,601.6,66.2z"/>
+				<path class="st0" d="M631.2,51c4.5,0,8.2,1.7,10.4,5.4c0.4,0.8,0.1,1.6-0.6,2.1l-1.5,1.3c-0.8,0.7-1.6,0.3-2.1-0.4
+					c-1.3-2.1-3.4-3.7-6.3-3.7c-5.9,0-11.7,5.5-11.7,14.3c0,5.7,2.9,10.8,8.1,10.8c4.1,0,6.8-3.5,7.8-5c0.6-0.9,1.4-1.1,2.1-0.5
+					l1.7,1.2c0.7,0.5,0.6,1.3,0.1,2.1c-2.1,3.7-6.6,6.9-12.4,6.9c-7.5,0-13-6.4-13-14.8C613.8,59.1,621.6,51,631.2,51z"/>
+				<path class="st0" d="M646.2,82.1c-0.4-0.4-0.6-1.1-0.1-1.8l1.2-1.9c0.4-0.7,1.3-0.8,2-0.2c1.2,1,3.8,2.6,6.8,2.6
+					c3.1,0,5.4-2.1,5.4-4.8c0-2.3-1.9-4-5.6-6.3c-3.4-2.1-7.2-4.4-7.2-9.1c0-4.2,3-9.7,10.9-9.7c4,0,6.7,1.4,8.1,2.4
+					c0.6,0.6,0.8,1.4,0.3,2.4l-0.7,1.4c-0.6,1-1.5,0.9-2.2,0.4c-1.9-1.2-3.6-1.9-5.9-1.9c-3.6,0-5,2.3-5,4.3c0,2.4,2.2,3.7,4.7,5.3
+					c4.3,2.7,8.1,5.2,8.1,10.2s-3.9,10-11.5,10C651.2,85.7,647.9,83.6,646.2,82.1z"/>
+			</g>
+		</g>
+	</g>
+</g>
+</svg>
diff --git a/src/vitepress/components/VPContent.vue b/src/vitepress/components/VPContent.vue
index 3628be3..72b2baf 100644
--- a/src/vitepress/components/VPContent.vue
+++ b/src/vitepress/components/VPContent.vue
@@ -1,22 +1,20 @@
 <script lang="ts" setup>
-import { useData, useRoute } from 'vitepress'
-import { useSidebar } from '../composables/sidebar'
-import VPContentPage from './VPContentPage.vue'
-import VPContentDoc from './VPContentDoc.vue'
-import VPNotFound from './VPNotFound.vue'
+import { useData, useRoute } from "vitepress";
+import { useSidebar } from "../composables/sidebar";
+import VPContentPage from "./VPContentPage.vue";
+import VPContentDoc from "./VPContentDoc.vue";
+import VPNotFound from "./VPNotFound.vue";
+import { useCopyCode } from "../composables/copy-code";
 
-const route = useRoute()
-const { frontmatter } = useData()
-const { hasSidebar } = useSidebar()
+const route = useRoute();
+const { frontmatter } = useData();
+const { hasSidebar } = useSidebar();
 
+useCopyCode();
 </script>
 
 <template>
-  <div
-    id="VPContent"
-    class="VPContent"
-    :class="{ 'has-sidebar': hasSidebar }"
-  >
+  <div id="VPContent" class="VPContent" :class="{ 'has-sidebar': hasSidebar }">
     <VPNotFound v-if="route.component === VPNotFound" />
     <VPContentPage v-else-if="!!frontmatter.page">
       <template #footer-before><slot name="footer-before" /></template>
diff --git a/src/vitepress/components/VPContentDoc.vue b/src/vitepress/components/VPContentDoc.vue
index 8d20c55..a2c11e8 100644
--- a/src/vitepress/components/VPContentDoc.vue
+++ b/src/vitepress/components/VPContentDoc.vue
@@ -1,20 +1,47 @@
 <script lang="ts" setup>
 import { computed } from "vue";
-import { useData } from "vitepress";
+import {useData, useRoute} from "vitepress";
 import VPContentDocOutline from "./VPContentDocOutline.vue";
 import VPContentDocFooter from "./VPContentDocFooter.vue";
 import type { Config } from "../config";
-import { VTLink, VTIconEdit } from "../../core";
+import { VTLink, VTIconGitHub, VTIconStackOverflow } from "../../core";
 
 const { page, frontmatter, theme } = useData<Config>();
 
-const hashMatch = /#(\w+)$/;
+const route = useRoute();
+
+const getMatchedRepos = (items) => {
+  return items.reduce((reduced, item) => {
+    // compare nav items with defined link and repo
+    if (item.link && item.repo && route.path.match(`^${item.link}`)) {
+      reduced.push({
+        repo: item.repo,
+        mount: item.mount,
+      });
+    }
+
+    // check for sub-items, deep-first
+    if (item.items) {
+      reduced = [
+        ...getMatchedRepos(item.items),
+        ...reduced,
+      ];
+    }
+
+    return reduced;
+  }, []);
+}
 
 const repoUrl = computed(() => {
-  const repo =
-    theme.value.editLink?.repo || "shopware/developer-documentation-vuepress";
-  const branch = repo.match(hashMatch)?.[1] || "main";
-  return `https://github.com/${repo}/edit/${branch}/src/${page.value.relativePath}`;
+  const matchedRepo = getMatchedRepos(theme.value.nav)[0];
+  const repo = matchedRepo?.repo
+      || theme.value.editLink?.repo
+      || "shopware/developer-documentation-vuepress";
+
+  const branch = repo.match(/#(\w+)$/)?.[1] || "main";
+  const folder = matchedRepo?.mount
+      || 'src';
+  return `https://github.com/${repo}/edit/${branch}/${folder}/${page.value.relativePath}`;
 });
 
 const pageClass = computed(() => {
@@ -43,15 +70,26 @@ const pageClass = computed(() => {
         <slot name="content-top" />
         <main>
           <Content class="vt-doc" :class="pageClass" />
+
           <p
             class="edit-link"
             v-if="theme.editLink && frontmatter.editLink !== false"
           >
-            <VTIconEdit class="vt-icon" />
+            <VTIconGitHub class="vt-icon" />
             <VTLink :href="repoUrl" :no-icon="true">{{
               theme.editLink.text
             }}</VTLink>
           </p>
+
+          <p
+            class="edit-link"
+            v-if="theme.editLink && frontmatter.stackOverflowLink !== false"
+          >
+            <VTIconStackOverflow class="vt-icon" />
+            <VTLink :href="'https://stackoverflow.com/questions/ask?tags=shopware'"
+                    :no-icon="true">Ask a question on StackOverflow</VTLink>
+          </p>
+
         </main>
         <slot name="content-bottom" />
         <VPContentDocFooter v-if="frontmatter.footer !== false" />
@@ -71,7 +109,7 @@ const pageClass = computed(() => {
 
 .content {
   margin: 0 auto;
-  max-width: 688px;
+  max-width: 768px; /* was 688px */
   position: relative;
 }
 
@@ -95,9 +133,12 @@ const pageClass = computed(() => {
 }
 
 .edit-link {
-  margin: 0 0 32px;
+  margin: 0 0 16px;
   /* text-align: center; */
 }
+.edit-link:last-child {
+  margin: 0 0 32px;
+}
 
 .edit-link .vt-link {
   font-size: 14px;
diff --git a/src/vitepress/components/VPNavBar.vue b/src/vitepress/components/VPNavBar.vue
index cf273ed..acd4931 100644
--- a/src/vitepress/components/VPNavBar.vue
+++ b/src/vitepress/components/VPNavBar.vue
@@ -39,7 +39,8 @@ defineProps<{
   border-bottom: 1px solid var(--vt-c-divider-light);
   padding: 0 12px 0 24px;
   height: var(--vt-nav-height);
-  background-color: var(--vt-c-bg);
+  background-color: var(--sw-nav-bg);
+  backdrop-filter: blur(12px);
   transition: border-color 0.5s, background-color 0.5s;
 }
 
diff --git a/src/vitepress/components/VPNavBarMenuLink.vue b/src/vitepress/components/VPNavBarMenuLink.vue
index 5af9d70..206c1e9 100644
--- a/src/vitepress/components/VPNavBarMenuLink.vue
+++ b/src/vitepress/components/VPNavBarMenuLink.vue
@@ -35,12 +35,13 @@ const { page } = useData()
   line-height: calc(var(--vt-nav-height) - 1px);
   font-size: 13px;
   font-weight: 500;
-  color: var(--vt-c-text-1);
+  color: #eee; /* Shopware Theme */
   transition: color 0.25s;
 }
 
 .VPNavBarMenuLink.active {
   border-bottom: 1px solid var(--vt-c-brand);
+  color: var(--vt-c-brand);
 }
 
 .VPNavBarMenuLink:hover {
diff --git a/src/vitepress/components/VPNavBarSearch.vue b/src/vitepress/components/VPNavBarSearch.vue
index 55b2e2b..95208eb 100644
--- a/src/vitepress/components/VPNavBarSearch.vue
+++ b/src/vitepress/components/VPNavBarSearch.vue
@@ -158,7 +158,7 @@ function load() {
 }
 
 .DocSearch-Button .DocSearch-Search-Icon {
-  color: var(--vt-c-text-2);
+  color: var(--vt-c-text-dark-2);
   transition: color 0.5s;
   fill: currentColor;
   width: 18px;
@@ -176,14 +176,14 @@ function load() {
 }
 
 .DocSearch-Button:hover .DocSearch-Search-Icon {
-  color: var(--vt-c-text-1);
+  color: var(--vt-c-brand); /* Shopware Theme */
 }
 
 .DocSearch-Button-Placeholder {
   transition: color 0.5s;
   font-size: 13px;
   font-weight: 500;
-  color: var(--vt-c-text-2);
+  color: var(--vt-c-text-dark-2); /* Shopware Theme */
   display: none;
   padding: 0 10px 0 0;
 }
@@ -195,12 +195,12 @@ function load() {
 }
 
 .DocSearch-Button:hover .DocSearch-Button-Placeholder {
-  color: var(--vt-c-text-1);
+  color: var(--vt-c-brand); /* Shopware Theme */
 }
 
 .DocSearch-Button .DocSearch-Button-Key {
   margin-top: 2px;
-  border: 1px solid var(--vt-c-divider);
+  border: 1px solid var(--vt-c-text-dark-2); /* Shopware Theme */
   border-right: none;
   border-radius: 4px 0 0 4px;
   display: none;
@@ -212,7 +212,7 @@ function load() {
 }
 
 .DocSearch-Button .DocSearch-Button-Key + .DocSearch-Button-Key {
-  border-right: 1px solid var(--vt-c-divider);
+  border-right: 1px solid var(--vt-c-text-dark-2); /* Shopware Theme */
   border-left: none;
   border-radius: 0 4px 4px 0;
   padding-left: 2px;
@@ -236,7 +236,7 @@ function load() {
   height: 20px;
   margin: 0;
   width: auto;
-  color: var(--vt-c-text-3);
+  color: var(--vt-c-text-dark-2);
   transition: color 0.5s;
   display: inline-block;
   padding: 0 1px;
diff --git a/src/vitepress/components/VPNavBarTitle.vue b/src/vitepress/components/VPNavBarTitle.vue
index d7b34a5..f52b562 100644
--- a/src/vitepress/components/VPNavBarTitle.vue
+++ b/src/vitepress/components/VPNavBarTitle.vue
@@ -1,19 +1,37 @@
 <template>
-  <a class="VPNavBarTitle" href="/">
-    <img src="/img/logo.svg" alt="Shopware" class="w-36px h-36px mr-2" />
-    <span class="text-lg font-bold">Shopware Docs</span>
-  </a>
+  <div class="flex items-center">
+    <a class="VPNavBarTitle" :href="withBase('/')">
+      <img
+        src="./../assets/shopware-docs.svg"
+        alt="Shopware"
+        class="VPNavBarImage"
+      />
+    </a>
+    <a :href="withBase('/installation.html')"
+      ><span class="accent text-xl">EA</span></a
+    >
+  </div>
 </template>
 
+<script setup>
+import { withBase } from "vitepress";
+</script>
+
 <style scoped>
 .VPNavBarTitle {
   display: flex;
   align-items: center;
   padding-top: 1px;
   height: var(--vt-nav-height);
+  color: var(--vt-c-brand);
   transition: opacity 0.25s;
 }
 
+.VPNavBarImage {
+  width: 90%;
+  height: 40px;
+}
+
 .VPNavBarTitle:hover {
   opacity: 0.6;
 }
diff --git a/src/vitepress/components/VPSidebar.vue b/src/vitepress/components/VPSidebar.vue
index 9441162..020515b 100644
--- a/src/vitepress/components/VPSidebar.vue
+++ b/src/vitepress/components/VPSidebar.vue
@@ -1,42 +1,105 @@
 <script lang="ts" setup>
-import { nextTick, ref, watchPostEffect } from 'vue'
-import { useSidebar } from '../composables/sidebar'
-import VPSidebarGroup from './VPSidebarGroup.vue'
+import { computed } from "@vue/reactivity";
+import { useData } from "vitepress";
+import { nextTick, ref, watchEffect, watchPostEffect } from "vue";
+import { useSidebar } from "../composables/sidebar";
+import { isPartiallyActive } from "../support/utils";
+import VPSidebarGroup from "./VPSidebarGroup.vue";
+import { SidebarGroup } from "../config";
+import { MenuItemWithLink } from "src/core";
 
-const { sidebar, hasSidebar } = useSidebar()
+const { sidebar, hasSidebar } = useSidebar();
 
 const props = defineProps<{
-  open: boolean
-}>()
+  open: boolean;
+}>();
 
 // A11y: Focus Nav element when menu has opened
-let navEl = ref<(Element & { focus(): void }) | null>(null)
+let navEl = ref<(Element & { focus(): void }) | null>(null);
 watchPostEffect(async () => {
   if (props.open) {
-    await nextTick()
-    navEl.value?.focus()
+    await nextTick();
+    navEl.value?.focus();
   }
-})
+});
+
+const { page } = useData();
+function hasActiveLink(items: SidebarGroup["items"]) {
+  const { relativePath } = page.value;
+  return items.some((item) => isPartiallyActive(relativePath, item.link));
+}
+
+const currentActiveGroup = ref();
+const currentActiveGroupElement = ref();
+const shouldShowAdditionalMenu = ref();
+
+watchEffect(async () => {
+  currentActiveGroup.value = sidebar.value.findLast((group: SidebarGroup) =>
+    hasActiveLink(group.items)
+  );
+  currentActiveGroupElement.value = currentActiveGroup.value?.items?.find(
+    (item: MenuItemWithLink) =>
+      isPartiallyActive(page.value.relativePath, item.link)
+  );
+  shouldShowAdditionalMenu.value = !!currentActiveGroupElement.value?.items?.length;
+});
 </script>
 
 <template>
   <aside
     v-if="hasSidebar"
     ref="navEl"
-    class="VPSidebar"
-    :class="{ open }"
+    class="VPSidebar flex"
+    :class="{ open, 'has-additional-menu': shouldShowAdditionalMenu }"
     @click.stop
   >
-    <nav id="VPSidebarNav" aria-labelledby="sidebar-aria-label" tabindex="-1">
-      <slot name="top" />
-      <span id="sidebar-aria-label" class="visually-hidden"
-        >Sidebar Navigation</span
+    <div class="VPSidebarNav__wrapper">
+      <nav
+        class="VPSidebarNav"
+        aria-labelledby="sidebar-aria-label"
+        tabindex="-1"
+      >
+        <slot name="top" />
+        <span id="sidebar-aria-label" class="visually-hidden"
+          >Sidebar Navigation</span
+        >
+        <div v-for="group in sidebar" :key="group.text" class="group">
+          <VPSidebarGroup
+            :link="group.link"
+            :text="group.text"
+            :items="group.items"
+            :showPartiallyActive="
+              currentActiveGroup && currentActiveGroup.text === group.text
+            "
+          />
+        </div>
+        <slot name="bottom" />
+      </nav>
+      <nav
+        class="VPSidebarNav"
+        aria-labelledby="sidebar-aria-label"
+        tabindex="-1"
+        v-if="currentActiveGroupElement"
       >
-      <div v-for="group in sidebar" :key="group.text" class="group">
-        <VPSidebarGroup :text="group.text" :items="group.items" />
-      </div>
-      <slot name="bottom" />
-    </nav>
+        <span id="sidebar-aria-label" class="visually-hidden"
+          >{{ currentActiveGroupElement.text }} section navigation</span
+        >
+        <h2
+          v-if="shouldShowAdditionalMenu"
+          class="font-bold pb-3 uppercase"
+        >
+          {{ currentActiveGroupElement.text }}
+        </h2>
+        <div
+          v-if="shouldShowAdditionalMenu"
+          v-for="group in currentActiveGroupElement.items"
+          :key="group.text"
+          class="group"
+        >
+          <VPSidebarGroup :text="group.text" :items="group.items" />
+        </div>
+      </nav>
+    </div>
   </aside>
 </template>
 
@@ -51,7 +114,7 @@ watchPostEffect(async () => {
   width: calc(100vw - 64px);
   max-width: var(--vp-sidebar-width-mobile);
   opacity: 0;
-  background-color: var(--vt-c-bg);
+  background-color: var(--sw-sidebar-bg);
   box-shadow: var(--vt-c-shadow-3);
   overflow-x: hidden;
   overflow-y: auto;
@@ -61,10 +124,13 @@ watchPostEffect(async () => {
   /* scrollbar-width: none; */
 }
 
-#VPSidebarNav {
+.VPSidebarNav {
   padding-top: 24px;
   outline: 0;
 }
+.VPSidebarNav__wrapper {
+  position: relative;
+}
 
 /* .VPSidebar::-webkit-scrollbar {
   display: none;
@@ -74,7 +140,7 @@ watchPostEffect(async () => {
   .VPSidebar {
     top: calc(var(--vt-nav-height) + var(--vt-banner-height, 0px));
     z-index: 1;
-    border-right: 1px solid var(--vt-c-divider-light);
+    border-right: 1px solid var(--sw-sidebar-border);
     width: var(--vp-sidebar-width-small);
     max-width: 100%;
     opacity: 1;
@@ -85,12 +151,33 @@ watchPostEffect(async () => {
   }
 }
 
-@media (min-width: 1440px) {
+@media (min-width: 1620px) {
   .VPSidebar {
     padding: 0 32px 96px calc((100% - var(--vp-screen-max-width)) / 2);
     width: calc(
       (100% - var(--vp-screen-max-width)) / 2 + var(--vp-sidebar-width-small)
     );
+    transition: padding 0.3s linear;
+  }
+
+  .has-additional-menu {
+    --nav-left: -100px;
+  }
+  .VPSidebarNav:nth-child(1) {
+    position: relative;
+    left: var(--nav-left, 0);
+    transition: left 0.5s cubic-bezier(0.19, 1, 0.22, 1);
+  }
+
+  .VPSidebarNav:nth-child(2) {
+    position: absolute;
+    left: 200%;
+    top: 0;
+    width: 100%;
+    transition: left 0.6s cubic-bezier(0.19, 1, 0.22, 1);
+  }
+  .has-additional-menu .VPSidebarNav:nth-child(2) {
+    left: 50%;
   }
 }
 
diff --git a/src/vitepress/components/VPSidebarGroup.vue b/src/vitepress/components/VPSidebarGroup.vue
index 9f9014f..66cbde6 100644
--- a/src/vitepress/components/VPSidebarGroup.vue
+++ b/src/vitepress/components/VPSidebarGroup.vue
@@ -1,18 +1,27 @@
 <script lang="ts" setup>
-import { MenuItemWithLink } from '../../core'
-import VPSidebarLink from './VPSidebarLink.vue'
-import { isActive } from '../support/utils'
-import { useData } from 'vitepress'
+import { MenuItemWithLink } from "../../core";
+import VPSidebarLink from "./VPSidebarLink.vue";
+import { isActive, isPartiallyActive } from "../support/utils";
+import { useData } from "vitepress";
 
 const props = defineProps<{
-  text: string
-  items: MenuItemWithLink[]
-}>()
+  link: string,
+  text: string;
+  items: MenuItemWithLink[];
+  showPartiallyActive?: boolean;
+}>();
 
-const { page } = useData()
+function activeMethod(currentPath: string, matchPath: string) {
+  if (props.showPartiallyActive) {
+    return isPartiallyActive(currentPath, matchPath);
+  }
+  return isActive(currentPath, matchPath);
+}
+
+const { page } = useData();
 function hasActiveLink() {
-  const { relativePath } = page.value
-  return props.items.some((item) => isActive(relativePath, item.link))
+  const { relativePath } = page.value;
+  return props.items.some((item) => activeMethod(relativePath, item.link));
 }
 </script>
 
@@ -20,12 +29,17 @@ function hasActiveLink() {
   <section class="VPSidebarGroup">
     <div class="title">
       <h2 class="title-text" :class="{ active: hasActiveLink() }">
-        {{ text }}
+        <VPSidebarLink
+            v-if="link"
+            :item="{text, link}"
+            :showPartiallyActive="showPartiallyActive"
+            :link-class="null"/>
+        <template v-else>{{ text }}</template>
       </h2>
     </div>
 
     <template v-for="item in items" :key="item.link">
-      <VPSidebarLink :item="item" />
+      <VPSidebarLink :item="item" :showPartiallyActive="showPartiallyActive" />
     </template>
   </section>
 </template>
@@ -47,5 +61,6 @@ function hasActiveLink() {
   font-weight: 600;
   color: var(--vt-c-text-1);
   transition: color 0.5s;
+  text-transform: uppercase;
 }
 </style>
diff --git a/src/vitepress/components/VPSidebarLink.vue b/src/vitepress/components/VPSidebarLink.vue
index cbc116e..8d409f4 100644
--- a/src/vitepress/components/VPSidebarLink.vue
+++ b/src/vitepress/components/VPSidebarLink.vue
@@ -1,24 +1,35 @@
 <script lang="ts" setup>
-import { useData } from 'vitepress'
-import { inject } from 'vue'
-import { MenuItemWithLink } from '../../core'
-import { isActive } from '../support/utils'
+import { useData } from "vitepress";
+import { inject } from "vue";
+import { MenuItemWithLink } from "../../core";
+import { isActive, isPartiallyActive } from "../support/utils";
 
-defineProps<{
-  item: MenuItemWithLink
-}>()
+const props = withDefaults(defineProps<{
+  item: MenuItemWithLink;
+  showPartiallyActive?: boolean;
+  linkClass?: string;
+}>(), {
+  linkClass: 'link-text'
+});
 
-const { page } = useData()
-const closeSideBar = inject('close-sidebar') as () => void
+const { page } = useData();
+const closeSideBar = inject("close-sidebar") as () => void;
+
+function activeMethod(currentPath: string, matchPath: string) {
+  if (props.showPartiallyActive) {
+    return isPartiallyActive(currentPath, matchPath);
+  }
+  return isActive(currentPath, matchPath);
+}
 </script>
 
 <template>
   <a
-    :class="{ link: true, active: isActive(page.relativePath, item.link) }"
+    :class="{ link: true, active: activeMethod(page.relativePath, item.link) }"
     :href="item.link"
     @click="closeSideBar"
   >
-    <p class="link-text">{{ item.text }}</p>
+    <p :class="linkClass">{{ item.text }}</p>
   </a>
 </template>
 
@@ -39,7 +50,8 @@ const closeSideBar = inject('close-sidebar') as () => void
   transition: color 0.25s;
 }
 
-.link.active .link-text {
+.link.active .link-text,
+.title-text .link.active {
   font-weight: 600;
   color: var(--vt-c-brand);
   transition: color 0.25s;
diff --git a/src/vitepress/composables/copy-code.ts b/src/vitepress/composables/copy-code.ts
new file mode 100644
index 0000000..c78a35b
--- /dev/null
+++ b/src/vitepress/composables/copy-code.ts
@@ -0,0 +1,94 @@
+import { nextTick, watch } from "vue";
+import { inBrowser, useData } from "vitepress";
+
+export function useCopyCode() {
+  const { page } = useData();
+
+  if (inBrowser)
+    watch(
+      () => page.value.relativePath,
+      () => {
+        nextTick(() => {
+          document
+            .querySelectorAll<HTMLSpanElement>(
+              '.vt-doc div[class*="language-"] > button.copy'
+            )
+            .forEach(handleElement);
+        });
+      },
+      { immediate: true, flush: "post" }
+    );
+}
+
+async function copyToClipboard(text: string) {
+  try {
+    return navigator.clipboard.writeText(text);
+  } catch {
+    const element = document.createElement("textarea");
+    const previouslyFocusedElement = document.activeElement;
+
+    element.value = text;
+
+    // Prevent keyboard from showing on mobile
+    element.setAttribute("readonly", "");
+
+    element.style.contain = "strict";
+    element.style.position = "absolute";
+    element.style.left = "-9999px";
+    element.style.fontSize = "12pt"; // Prevent zooming on iOS
+
+    const selection = document.getSelection();
+    const originalRange = selection
+      ? selection.rangeCount > 0 && selection.getRangeAt(0)
+      : null;
+
+    document.body.appendChild(element);
+    element.select();
+
+    // Explicit selection workaround for iOS
+    element.selectionStart = 0;
+    element.selectionEnd = text.length;
+
+    document.execCommand("copy");
+    document.body.removeChild(element);
+
+    if (originalRange) {
+      selection!.removeAllRanges(); // originalRange can't be truthy when selection is falsy
+      selection!.addRange(originalRange);
+    }
+
+    // Get the focus back on the previously focused element, if any
+    if (previouslyFocusedElement) {
+      (previouslyFocusedElement as HTMLElement).focus();
+    }
+  }
+}
+
+function handleElement(el: HTMLElement) {
+  el.onclick = () => {
+    const parent = el.parentElement;
+    const sibling = el.nextElementSibling
+      ?.nextElementSibling as HTMLPreElement | null;
+    if (!parent || !sibling) {
+      return;
+    }
+
+    const isShell = /language-(shellscript|shell|bash|sh|zsh)/.test(
+      parent.classList.toString()
+    );
+
+    let { innerText: text = "" } = sibling;
+
+    if (isShell) {
+      text = text.replace(/^ *(\$|>) /gm, "");
+    }
+
+    copyToClipboard(text).then(() => {
+      el.classList.add("copied");
+      setTimeout(() => {
+        el.classList.remove("copied");
+        el.blur();
+      }, 2000);
+    });
+  };
+}
diff --git a/src/vitepress/config.ts b/src/vitepress/config.ts
index 7466078..f7446ec 100644
--- a/src/vitepress/config.ts
+++ b/src/vitepress/config.ts
@@ -99,6 +99,13 @@ export interface MultiSidebarConfig {
 }
 
 export interface SidebarGroup {
+  link?: string;
   text: string;
-  items: MenuItemWithLink[];
+  items: Array<MenuItemWithLink | AdditionalMenuItemWithContext>;
 }
+
+export type AdditionalMenuItemWithContext = {
+  text: string;
+  link: string;
+  items: SidebarGroup[];
+};
diff --git a/src/vitepress/config/baseConfig.js b/src/vitepress/config/baseConfig.js
index 43c7e96..2dad47d 100644
--- a/src/vitepress/config/baseConfig.js
+++ b/src/vitepress/config/baseConfig.js
@@ -16,13 +16,19 @@ const {
   transformerVariantGroup,
 } = require("unocss");
 
+// remove navigation from the library
+// const navigation = require("./navigation");
+const navigation = [];
+
+const { withMermaid } = require("vitepress-plugin-mermaid");
+
 // for local-linked development
 const deps = ["vitepress-shopware-docs", "@vueuse/core", "body-scroll-lock"];
 
 /**
  * @type {() => Promise<import('vitepress').UserConfig>}
  */
-module.exports = async () => ({
+module.exports = async () => withMermaid({
   vite: {
     ssr: {
       noExternal: deps,
@@ -30,6 +36,18 @@ module.exports = async () => ({
     optimizeDeps: {
       exclude: deps,
     },
+    resolve: {
+      // for mounting static sub-repos in dev env
+      preserveSymlinks: true
+    },
+    // https://www.npmjs.com/package/@rollup/plugin-node-resolve ?
+    // https://github.com/vuejs/vitepress/blob/main/rollup.config.ts ?
+    build: {
+      rollupOptions: {
+        preserveSymlinks: true,
+        shimMissingExports: true,
+      },
+    },
     plugins: [
       Unocss.default(
         defineConfig({
@@ -57,13 +75,6 @@ module.exports = async () => ({
   },
 
   head: [
-    [
-      "link",
-      {
-        rel: "icon",
-        href: "/img/logo.svg",
-      },
-    ],
     ...(process.env.NODE_ENV === "production"
       ? [
           [
@@ -95,8 +106,27 @@ module.exports = async () => ({
     highlight: await require("./highlight")(),
   },
 
+  vue: {
+    reactivityTransform: true,
+    template: {
+      compilerOptions: {
+        isCustomElement: (tag) => tag.startsWith("elements-"),
+      },
+    },
+  },
+
   shouldPreload: (link) => {
     // make algolia chunk prefetch instead of preload
     return !link.includes("Algolia");
   },
+  themeConfig: {
+    nav: navigation,
+    appearance: true,
+    socialLinks: [
+      { icon: "github", link: "https://github.com/shopware/" },
+      { icon: "twitter", link: "https://twitter.com/ShopwareDevs" },
+      { icon: "slack", link: "https://slack.shopware.com" },
+      { icon: "stackoverflow", link: "https://stackoverflow.com/questions/tagged/shopware" },
+    ],
+  },
 });
diff --git a/src/vitepress/config/highlight.js b/src/vitepress/config/highlight.js
index c4dde9f..ef66796 100644
--- a/src/vitepress/config/highlight.js
+++ b/src/vitepress/config/highlight.js
@@ -10,15 +10,45 @@ function escapeHtml(html) {
   return html.replace(/[&<>"']/g, (chr) => htmlEscapes[chr])
 }
 
+function range(size, startAt = 0) {
+  return [...Array(size).keys()].map(i => i + startAt);
+}
+
+/**
+ * Translate the "1,2,6-8" into an array of numbers to be like: 1,2,6,7,8
+ */
+function getHighlightLinesConfig(linesConfig = '') {
+  if (!linesConfig) {
+    return;
+  }
+
+  let linesToHighlight = [];
+  const linesGroups = linesConfig.split(',')
+  linesGroups.forEach(group => {
+    const rangeParts = group.split('-')
+    if (rangeParts.length === 1) {
+      linesToHighlight.push(parseInt(rangeParts[0]))
+    } else {
+      const start = parseInt(rangeParts[0])
+      const end = parseInt(rangeParts[1])
+      linesToHighlight = linesToHighlight.concat(range(end - start + 1, start))
+    }
+  })
+
+  return linesToHighlight;
+}
+
 module.exports = async (theme = 'material-palenight') => {
   const highlighter = await require('shiki').getHighlighter({
     theme
   })
 
-  return (code, lang) => {
+  return (code, lang, attrs) => {
     if (!lang || lang === 'text') {
       return `<pre v-pre><code>${escapeHtml(code)}</code></pre>`
     }
-    return highlighter.codeToHtml(code, lang).replace(/^<pre.*?>/, '<pre v-pre>')
+    const linesConfig = getHighlightLinesConfig(attrs);
+    return highlighter.codeToHtml(code, {lang, lineOptions: linesConfig?.map(line => ({line, classes: ["highlighted"]}))} ).replace(/^<pre.*?>/, '<pre v-pre>')
+
   }
 }
diff --git a/src/vitepress/config/navigation.js b/src/vitepress/config/navigation.js
new file mode 100644
index 0000000..6f8a120
--- /dev/null
+++ b/src/vitepress/config/navigation.js
@@ -0,0 +1,70 @@
+const nav = [
+  {
+    activeMatch: "^/apps/",
+    text: "Apps",
+    link: '/apps/'
+  },
+  {
+    text: "Themes",
+    link: "/themes/",
+  },
+  {
+    text: "Frontends",
+    link: "/frontends/",
+  },
+  {
+    text: "Integrations",
+    link: "/integrations/",
+  },
+  {
+    text: "Resources",
+    activeMatch: `^/(api)`,
+    items: [
+      {
+          text: "HTTP APIs",
+          items: [
+              {
+                  text: "Store API",
+                  link: "/resources/api/store-api-reference",
+              },
+              {
+                  text: "Admin API",
+                  link: "/resources/api/admin-api-reference",
+              }
+          ]
+      },
+      {
+          text: "Administration",
+          items: [
+              {
+                  text: "Admin Extension API",
+                  link: "/resources/api/store-api-reference",
+              },
+              {
+                  text: "Meteor Icon Kit",
+                  link: "/resources/meteor-icon-kit",
+              },
+              {
+                  text: "Component Library",
+                  link: "/resources/meteor-icon-kit",
+              }
+          ]
+      },
+      {
+          text: "Learning",
+          items: [
+              {
+                  text: "Academy",
+                  link: "https://academy.shopware.com/",
+              },
+              {
+                text: "YouTube",
+                link: "https://www.youtube.com/user/shopwareAG"
+              }
+          ]
+      }
+    ]
+  },
+];
+
+module.exports = nav;
\ No newline at end of file
diff --git a/src/vitepress/index.ts b/src/vitepress/index.ts
index 39cc2bf..2c38f8b 100644
--- a/src/vitepress/index.ts
+++ b/src/vitepress/index.ts
@@ -6,9 +6,28 @@ import VPNotFound from "./components/VPNotFound.vue";
 import { Theme } from "vitepress";
 import { withConfigProvider } from "./composables/config";
 
+// import ActionItem from "./../shopware/components/ActionItem.vue";
+// import RegistrationForm from "./../shopware/components/RegistrationForm.vue";
+// import LandingWrapper from "./../shopware/components/LandingWrapper.vue";
+// import TopBar from "./../shopware/components/TopBar.vue";
+import PageRef from "./../shopware/components/PageRef.vue";
+import CodeBlock from "./../shopware/components/CodeBlock.vue";
+import Tabs from "./../shopware/components/Tabs.vue";
+import Tab from "./../shopware/components/Tab.vue";
+
 const VPTheme: Theme = {
   Layout: withConfigProvider(VPApp),
   NotFound: VPNotFound,
+  enhanceApp({ app }) {
+    // app.component('ActionItem', ActionItem),
+    // app.component('RegistrationForm', RegistrationForm),
+    // app.component('LandingWrapper', LandingWrapper),
+    // app.component('TopBar', TopBar),
+    app.component("PageRef", PageRef),
+      app.component("CodeBlock", CodeBlock),
+      app.component("Tabs", Tabs),
+      app.component("Tab", Tab);
+  },
 };
 
 export { VPTheme };
diff --git a/src/vitepress/styles/variables.css b/src/vitepress/styles/variables.css
index 164607f..d595af3 100644
--- a/src/vitepress/styles/variables.css
+++ b/src/vitepress/styles/variables.css
@@ -12,6 +12,10 @@
 
   /* Screen Size */
   --vp-screen-max-width: 1376px;
+
+  /* Colors */
+  --vp-code-line-highlight-color: rgba(0, 0, 0, 0.5);
+
 }
 
 /**
diff --git a/src/vitepress/support/utils.ts b/src/vitepress/support/utils.ts
index f425ef6..6af4d56 100644
--- a/src/vitepress/support/utils.ts
+++ b/src/vitepress/support/utils.ts
@@ -56,6 +56,18 @@ export function isActive(
   }
 }
 
+export function isPartiallyActive(
+  currentPath: string,
+  matchPath?: string
+): boolean {
+  if (matchPath === undefined) {
+    return false;
+  }
+  currentPath = normalize(`/${currentPath}`);
+  matchPath = normalize(`${matchPath}`);
+  return currentPath.startsWith(matchPath);
+}
+
 export function normalize(path: string): string {
   return decodeURI(path).replace(hashRE, "").replace(extRE, "");
 }
